firefox-ios/Client/Frontend/Browser/MainMenu/MainMenuConfigurationUtility.swift

```diff
@@ -66,78 +72,99 @@ struct MainMenuConfigurationUtility: Equatable {
         ])
     }
 
-    private func getToolsSection(with uuid: WindowUUID) -> MenuSection {
-        return MenuSection(options: [
-            MenuElement(
-                title: .MainMenu.ToolsSection.SwitchToDesktopSite,
-                iconName: "",
-                isEnabled: true,
-                isActive: false,
-                a11yLabel: "",
-                a11yHint: "",
-                a11yId: "",
-                action: {
-                    store.dispatch(
-                        MainMenuAction(
-                            windowUUID: uuid,
-                            actionType: MainMenuActionType.closeMenu
+    private func getToolsSection(
+        with uuid: WindowUUID,
+        and configuration: MainMenuTabInfo?
+    ) -> MenuSection {
+        guard let configuration else { return MenuSection(options: []) }
```

Comment:
There will be a case where this is `nil` yes. It will be when the menu is initialized. Because we don't have the info yet, it's empty. Redux then calls out to get the info on viewDidLoad, and updates the state with the required tab info, and then we rebuild the menu elements and reload the tableview.

Reply:
thanks! so are we expecting that on initial load that we won't have this section? Or if we always want to section, then instead of rebuilding, why not only call this after menu is initialized?